lib/pharos/phases/configure_master.rb

```diff
@@ -95,19 +108,26 @@ def install
       end
 
       def generate_config
+        extra_sans = Set.new
+        @config.master_hosts.each do |h|
+          extra_sans += [h.address, h.private_address].compact.uniq
```

Comment:
Yes, I think it would also work. Anyway this logic needs improvement because admin probably should "mask" multiple masters with LB or DNS round-robin.